lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -113,35 +113,82 @@ inline bool _IsUsdUVTexture(const HdMaterialNode& node)
     return (node.identifier == UsdImagingTokens->UsdUVTexture);
 }
 
-//! Helper utility function to print nodes, connections and primvars in the
+//! Helper function to generate a XML string about nodes, relationships and primvars in the
 //! specified material network.
-void _PrintMaterialNetwork(
-    const std::string&       label,
-    const SdfPath&           id,
-    const HdMaterialNetwork& mat)
+std::string _GenerateXMLString(const HdMaterialNetwork& mat, bool includeParams=true)
 {
-    std::cout << label << " material network for " << id << "\n";
+    std::string result;
+    result.reserve(1024);
+
+    if (!mat.nodes.empty()) {
+        result += "<nodes>\n";
+
+        if (includeParams) {
+            for (HdMaterialNode const& node : mat.nodes) {
+                result += "  <node name=\"";
+                result += node.path.GetName();
```

Comment:
Not sure about this one. You definitely do not want the full path since it includes the material name and this would prevent useful merges. The problem I see is with ambiguous paths inside the material. Have you noticed how often the name "TexCoordReader" comes back in a Maya export? You probably need the relative path rooted at the material: something that truncates `/pSphere1/Looks/standardSurface2SG/file1/TexCoordReader.outputs` into `./file1/TexCoordReader`.

Reply:
Good point, Jerry. I never noticed the hierarchy in the material network. I will try to reproduce this scenario first. Thanks.